vault/resource_azure_secret_backend_role.go

```diff
@@ -90,13 +107,25 @@ func azureSecretBackendRoleUpdateFields(d *schema.ResourceData, data map[string]
 	if v, ok := d.GetOk("azure_roles"); ok {
 		rawAzureList := v.(*schema.Set).List()
 
-		// Vaults API requires we send trhe policy as an escaped string
+		// Vaults API requires we send the policy as an escaped string
 		// So we marshall and then change into a string
 		jsonAzureList, _ := json.Marshal(rawAzureList)
 		jsonAzureListString := string(jsonAzureList)
 
 		log.Printf("[DEBUG] Azure RoleSet turned to escaped JSON: %s", jsonAzureListString)
-		data["azure_roles"] = string(jsonAzureListString)
+		data["azure_roles"] = jsonAzureListString
+	}
+
+	if v, ok := d.GetOk("azure_groups"); ok {
+		rawAzureList := v.(*schema.Set).List()
+
+		// Vaults API requires we send the policy as an escaped string
+		// So we marshall and then change into a string
+		jsonAzureList, _ := json.Marshal(rawAzureList)
```

Comment:
This error should be handled incase there's a marshaling error (HCL to JSON can sometimes cause unexpected behavior).